### ISSUE TITLE
Check fails on parameter issues, but they don't seem to exist

### ISSUE THREAD

**Issue by 4c0n** [astral-sh/ty#3075]
### Summary

Hi,

Thanks for this great tool, we're very happy with the performance!

We seem to be running into a false-positive here:
https://github.com/Amsterdam/ee-rfc9068-python-access-token-validator/pull/39/
https://github.com/Amsterdam/ee-rfc9068-python-access-token-validator/actions/runs/23246402770/job/67575889418

```
Run uv run ty check --output-format github
Error: src/rfc9068/signature.py:25:15: missing-argument: No argument provided for required parameter `kid` of function `get_signing_key`
  info: Union variant `def get_signing_key(self, kid: str) -> PyJWK` is incompatible with this call site
  info: Attempted to call union type `(bound method PyJWKClient.get_signing_key(kid: str) -> PyJWK) | (def get_signing_key(self, kid: str) -> PyJWK)`
  info: rule `missing-argument` is enabled by default
Error: src/rfc9068/signature.py:25:49: invalid-argument-type: Argument to function `get_signing_key` is incorrect: Expected `PyJWKClient`, found `str`
  .venv/lib/python3.14/site-packages/jwt/jwks_client.py:172:9: info: Function defined here
    .venv/lib/python3.14/site-packages/jwt/jwks_client.py:172:25: Parameter declared here
  info: Union variant `def get_signing_key(self, kid: str) -> PyJWK` is incompatible with this call site
  info: Attempted to call union type `(bound method PyJWKClient.get_signing_key(kid: str) -> PyJWK) | (def get_signing_key(self, kid: str) -> PyJWK)`
  info: rule `invalid-argument-type` is enabled by default
Error: Process completed with exit code 1.
```

The first one does not seem to make much sense, as the `kid` parameter is provided.
The second one makes even less sense to me, as even the signature snippet it prints states that it expects a `str` type, not a `PyJWKClient`.

Hopefully this report helps improve `ty` and make it even greater :)
If we're doing something wrong please let me know as well.

### Version

ty 0.0.23

**Comment by carljm** [comment]
Hi, thanks for the report!

The problem is that ty thinks the type of the method you are calling is the union `(bound method PyJWKClient.get_signing_key(kid: str) -> PyJWK) | (def get_signing_key(self, kid: str) -> PyJWK)`. The first element of that union is fine. The second element we are not treating as a bound method, but as a function with two parameters (`self` and `kid`), so the single argument you pass positionally is assigned to `self`, leaving no argument provided for `kid`.

The root cause of this problem is this line in the PyJWT library: https://github.com/jpadilla/pyjwt/blob/master/jwt/jwks_client.py#L93

This is tracked in #350.

**Comment by 4c0n** [comment]
Ah right, that makes total sense now. Thanks for explaining!
I get what they're trying to achieve, but it's a really dirty way of working. Nonetheless possible.
